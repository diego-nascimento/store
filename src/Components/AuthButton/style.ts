import styled from 'styled-components'

export const Wrapper = styled.div`
  margin-right: 10px;
  cursor: pointer;
  display: flex;
  justify-content: flex-end;
  width: fit-content;
  min-width: 150px;
`

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`

export const MessagesContainer = styled.div`
  p,
  b {
    padding: 0px;
    margin: 0px 0px 0px 5px;
    font-size: 0.8rem;
  }
`
