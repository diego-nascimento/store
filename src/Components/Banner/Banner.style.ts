import styled from 'styled-components'

export const Container = styled.div`
  width: 100%;
  position: relative;

  a{
    display: flex;
    justify-content: center;
  }

  a img{
    width: 100%;
  }
`
