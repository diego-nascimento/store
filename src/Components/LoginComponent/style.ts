import styled from 'styled-components'

export const Container = styled.div`
  width: 100%;
  max-width: 500px;
  padding: 20px 20px;
`

export const Formulario = styled.form`
  border: 1px solid #111;
  padding: 190px 15px;
  border-radius: 7px;
  display: flex;
  flex-direction: column;
`

export const ButtonContainer = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 20px;

  button {
    margin: 10px 10px;
  }
`

export const Options = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;

  p {
    cursor: pointer;
  }
`
