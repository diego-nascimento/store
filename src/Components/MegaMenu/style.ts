import styled from 'styled-components'

interface ContainerProps {
  MegaMenuState: boolean
}

export const Container = styled.nav<ContainerProps>`
  background: white;
  position: absolute;
  width: 80vw;
  max-width: 1000px;
  top: 40px;
  left: 10px;
  transition: .2s;
  visibility: ${props => props.MegaMenuState ? 'visible' : 'hidden'};
  transform:${props => props.MegaMenuState ? 'translateY(0px)' : 'translateY(-10px)'} ;
  opacity: ${props => props.MegaMenuState ? '1' : '0'} ;
  padding: 10px 10px;
  border-radius: 7px;
  border: 1px solid #999;
`

export const TitleContainer = styled.div`
  width: 100%;
  margin-bottom: 25px;
`

export const Title = styled.h3`
  text-transform: capitalize;
  font-size: 1rem;
`

export const CategoriasContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  width: 100%;
  grid-gap: 10px;

`
export const CategoriaItem = styled.div`
  width: 100%;

  p{
    font-size: .8rem;
    text-transform: capitalize;
    text-decoration: none;
    font-weight: 500;
    cursor: pointer;
  }
  
`
