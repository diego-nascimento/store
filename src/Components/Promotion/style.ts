import styled from 'styled-components'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: 20px 0px;

`

type PromotionProps = {
  amount: number
}

export const Promotion = styled.div<PromotionProps>`
  display: grid;
  grid-template-columns: ${props => `repeat(${props.amount}, 1fr)`};
  width: 100%;
  grid-gap: 20px;
  border-bottom: 1px solid #ccc;

  @media(max-width: 800px){
    grid-template-columns: 1fr;
  }
`

export const ImagemContainer = styled.div`
  width: 100%;
  transition: .3s;
  
  img{
    width: 100%;
    cursor: pointer;
    
    :hover{
      opacity: .8;
    }
  }
`

export const BottomContainer = styled.div`
   display: flex;
   flex-direction: row;
   justify-content: space-between;
   padding: 0px 5px;
   align-items: center;
`

export const InfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  padding: 10px 0px;

  b{
    font-size: 1rem;
    display: flex;
    align-items: center;
  }

  p{
    font-size: .9rem;
    color: #999;
    display: flex;
    align-items: center;
    padding: 0px;
    margin: 0px;
  }

`

export const Button = styled.div`
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #111111;
  padding: 10px 15px;
  text-transform: uppercase;
  font-weight: bold;
  cursor: pointer;

  p{
    padding: 0px;
    margin: 0px;
  }

  :hover{
    opacity: .8;
  }
`
