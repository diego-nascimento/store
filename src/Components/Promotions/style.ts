import styled from 'styled-components'

export const Container = styled.section`
  padding: 0px;
  margin: 0px;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`
