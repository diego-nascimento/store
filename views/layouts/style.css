h1{
  color: pink;
  font-size: 36px;
}