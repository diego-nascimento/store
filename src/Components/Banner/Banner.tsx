import React from 'react'
import { Carousel } from 'react-bootstrap'
import { Container } from './Banner.style'
import Link from 'next/link'
import { BannerType } from '../../typing/Interfaces/IHomePage'

interface IBanner {
  BannersData: Array<BannerType>
  mobile: boolean
}

const Banner: React.FC<IBanner> = ({ BannersData, mobile }) => {
  console.log(BannersData)
  return (
    BannersData && Array.isArray(BannersData) && BannersData.length > 0
      ? <Container className='Container'>
        <Carousel
          controls={true}
          touch={true}
          indicators={true}
          slide={true}
          interval={5000}

        >
          {BannersData.map((banner) => {
            return (
              <Carousel.Item key={banner.id}>
                {banner.imagem !== null &&
                  <Link href={banner.url}>
                    <a style={{ width: '100%', cursor: 'pointer' }} target='blank'>
                      <img src={mobile && banner.imagemMobile ? banner.imagemMobile.url : banner.imagem.url} alt={banner.alt} />
                    </a>
                  </Link>
                }

              </Carousel.Item>
            )
          })}
        </Carousel>
      </Container>
      : null
  )
}

export default Banner
