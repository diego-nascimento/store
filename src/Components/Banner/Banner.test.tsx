import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Banner from './Banner'
import React from 'react'
import { Container } from './Banner.style'

Enzyme.configure({ adapter: new Adapter() })

describe('BannerComponent', () => {
  test('BannerComponent should render with no errors', () => {
    const wrapper = shallow(<Banner />)
    expect(wrapper).toBeTruthy()
  })

  test('Should find a Container in Component', () => {
    const wrapper = shallow(<Banner />)
    expect(wrapper.find(<Container />)).toBeTruthy()
  })

  test('Should match snapshot', () => {
    const wrapper = shallow(<Banner />)
    expect(wrapper).toMatchSnapshot()
  })
})
