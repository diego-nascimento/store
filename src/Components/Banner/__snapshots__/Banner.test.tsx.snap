// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['BannerComponent Should match snapshot 1'] = 'ShallowWrapper {}'
